VSCode/armsnippets.json

```diff
@@ -506,6 +506,145 @@
+        "                          \"description\": \"${15:natRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${16:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"[reference(resourceId('Microsoft.Network/publicIPAddresses', '${34:publicIPAddress1}')).ipAddress]\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${18:3389}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${19|TCP,UDP,Any,ICMP|}\"",
+        "                          ],",
+        "                          \"translatedAddress\": \"${20:10.0.0.254}\",",
+        "                          \"translatedPort\": \"${21:2500}\"",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"networkRuleCollections\": [",
+        "           {",
+        "              \"name\": \"${22:networkCollectionName}\",",
+        "               \"properties\": {",
+        "                   \"priority\": \"${23:100}\",",
+        "                   \"action\": {",
+        "                      \"type\": \"${24|Deny,Allow|}\"",
+        "                   },",
+        "                   \"rules\": [",
+        "                      {",
+        "                          \"name\": \"${25:networkRuleName}\",",
+        "                          \"description\": \"${26:networkRuleDescription}\",",
+        "                          \"sourceAddresses\": [",
+        "                             \"${27:10.0.0.4}\"",
+        "                          ],",
+        "                          \"destinationAddresses\": [",
+        "                              \"${28:10.0.1.4}\"",
+        "                          ],",
+        "                          \"destinationPorts\": [",
+        "                              \"${29:1}\"",
+        "                          ],",
+        "                          \"protocols\": [",
+        "                              \"${30|TCP,UDP,Any,ICMP|}\"",
+        "                          ]",
+        "                      }",
+        "                   ]",
+        "               }",
+        "           }",
+        "        ],",
+        "       \"ipConfigurations\": [",
+        "           {",
+        "               \"name\": \"${31:ipConfigName}\",",
+        "               \"properties\": {",
+        "                   \"subnet\": {",
+        "                       \"id\": \"${[resourceId('virtualNetworkResourceGroup', 'Microsoft.Network/virtualNetworks/subnets', 'virtualNetworkName', 'AzureFirewallSubnet')]}\"",
```

Comment:
This resourceId() call is still wrapped in a placeholder

Reply:
{ facepalm }

Reply:
Looks like the whole resourceId function call is marked as a placeholder (\"id\": \"**${**[resourceId('vi...). Don't you want to tokenize the params passed to resourceId() instead?

Reply:
yup